gef.py

```diff
@@ -10482,22 +10482,86 @@ def lookup_command(self, cmd):
 
         return None
 
+@register_command
+class AliasesCommand(GenericCommand):
+    """Base command to add, remove, or list aliases"""
 
-class GefAliases(gdb.Command):
-    """List all custom aliases."""
+    _cmdline_ = "aliases"
+    _syntax_  = "{:s} (add|rm|list)".format(_cmdline_)
 
     def __init__(self):
-        super().__init__("aliases", gdb.COMMAND_OBSCURE, gdb.COMPLETE_NONE)
+        super().__init__(prefix=True)
         return
 
-    def invoke(self, args, from_tty):
-        self.dont_repeat()
+    def do_invoke(self, argv):
+        self.usage()
+        return
+
+@register_command
+class AliasesAddCommand(GenericCommand):
+    """Command to add aliases"""
+
+    _cmdline_ = "aliases add"
+    _syntax_  = "{0} [ALIAS] [COMMAND]".format(_cmdline_)
+    _example_ = "{0} scope telescope".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__(complete=gdb.COMPLETE_LOCATION)
```

Comment:
I don't think the completion class should be `gdb.COMPLETE_LOCATION`. I think `COMPLETE_COMMAND` makes more sense in this case (see https://sourceware.org/gdb/current/onlinedocs/gdb/Commands-In-Python.html#Commands-In-Python)